spot_wrapper/spot_graph_nav.py

```diff
@@ -0,0 +1,796 @@
+            upload_filepath = upload_path[:-1]
+        else:
+            upload_filepath = upload_path
+
+        # Boolean indicating the robot's power state.
+        power_state = self._robot_state_client.get_robot_state().power_state
+        self._started_powered_on = power_state.motor_power_state == power_state.STATE_ON
+        self._powered_on = self._started_powered_on
+
+        # Claim lease, power on robot, start graphnav.
+        self._lease = self._lease_wallet.get_lease()
+        self._lease_keepalive = LeaseKeepAlive(self._lease_client)
+        if upload_filepath:
+            self._clear_graph()
+            self._upload_graph_and_snapshots(upload_filepath)
+        else:
+            self._download_current_graph()
+            self._logger.info(
+                "Re-using existing graph on robot. Check that the correct graph is loaded!"
+            )
+        if initial_localization_fiducial:
+            self._set_initial_localization_fiducial()
+        if initial_localization_waypoint:
+            self._set_initial_localization_waypoint([initial_localization_waypoint])
+        self._list_graph_waypoint_and_edge_ids()
+        self._get_localization_state()
+
+        return True, "Localization done!"
+
+    def navigate_to_existing_waypoint(self, waypoint_id: str):
+        """Navigate to an existing waypoint.
+        Args:
+            waypoint_id : Waypoint id string for where to go
+        """
+        self._get_localization_state()
+        resp = self._navigate_to(waypoint_id)
+        return resp
+
+    def navigate_through_route(self, waypoint_ids: typing.List[str]):
+        """
+        Args:
+            waypoint_ids: List[str] of waypoints to follow
+        """
+        self._get_localization_state()
+        self._logger.info(f"Waypoint ids: {','.join(waypoint_ids)}")
+        resp = self._navigate_route(waypoint_ids)
+        return resp
+
+    def download_navigation_graph(self, download_path: str) -> typing.List[str]:
+        """Download the navigation graph.
+        Args:
+            download_path : Path to the root directory of the map.
+        """
+        self._download_filepath = download_path
+        self._download_full_graph()
+        return self.list_graph()
+
+    def navigation_close_loops(
+        self, close_fiducial_loops: bool, close_odometry_loops: bool
+    ) -> typing.Tuple[bool, str]:
+        return self._auto_close_loops(close_fiducial_loops, close_odometry_loops)
+
+    def optmize_anchoring(self) -> typing.Tuple[bool, str]:
+        return self._optimize_anchoring()
+
+    ## Copied from spot-sdk/python/examples/graph_nav_command_line/graph_nav_command_line.py
+    # Copyright (c) 2020 Boston Dynamics, Inc.  All rights reserved.
+    #
+    # Downloading, reproducing, distributing or otherwise using the SDK Software
+    # is subject to the terms and conditions of the Boston Dynamics Software
+    # Development Kit License (20191101-BDSDK-SL).
```

Comment:
```suggestion
    ## Copied from https://github.com/boston-dynamics/spot-sdk/blob/master/python/examples/graph_nav_command_line/recording_command_line.py and https://github.com/boston-dynamics/spot-sdk/blob/master/python/examples/graph_nav_command_line/graph_nav_command_line.py with minor modifications
    # Copyright (c) 2020 Boston Dynamics, Inc.  All rights reserved.
    #
    # Downloading, reproducing, distributing or otherwise using the SDK Software
    # is subject to the terms and conditions of the Boston Dynamics Software
    # Development Kit License (20191101-BDSDK-SL).
```

Should make clear that we have made some modifications

Reply:
Agreed, done in [392da89](https://github.com/bdaiinstitute/spot_wrapper/pull/6/commits/392da898d027fc716a17f1cd7ff487183cc192cb)